src/lib/comms/sol-netctl-impl-connman.c

```diff
@@ -797,6 +807,23 @@ sol_netctl_service_disconnect(struct sol_netctl_service *service)
         "net.connman.Service", "Disconnect", _service_disconnect, service, NULL);
 }
 
+static void
+_destory_agent_vector(struct sol_vector *vector)
+{
+    uint16_t i;
+    sol_netctl_agent_input *input;
+
+    if (!vector)
+        return;
+
+    SOL_VECTOR_FOREACH_IDX (vector, input, i) {
+        if (input->type)
+            free(input->type);
```

Comment:
if you never use `input->input` why we are keeping a vector of `sol_netctl_agent_input` ? we can keep only the input types (char *).